stan/math/prim/meta/scalar_type.hpp

```diff
@@ -53,11 +53,22 @@ struct scalar_type<T, std::enable_if_t<is_std_vector<T>::value>> {
  * Template metaprogram defining the base scalar type of
  * values stored in an Eigen matrix.
  *
- * @tparam T type of matrix.
+ * @tparam T value type of matrix
  */
 template <typename T>
 struct scalar_type<T, std::enable_if_t<is_eigen<T>::value>> {
   using type = scalar_type_t<typename std::decay_t<T>::Scalar>;
 };
+
+/** \ingroup type_trait
+ * Template metaprogram defining the base scalar type of
+ * values stored in a complex number.
+ *
+ * @tparam T value type of complex number
+ */
+template <typename T>
+struct scalar_type<std::complex<T>> {
```

Comment:
This could be a bit weird. The complex number is the scalar type. I feel like the way this is written it should be called autodiff_scalar_type.

return_type leans on scalar_type (https://github.com/stan-dev/math/blob/ed61d817eb71f36fa2cd35f75b7c65c3135a817a/stan/math/prim/meta/return_type.hpp#L33), and so ```return_type<std::complex<var>, double>``` would be ```var```, which isn't what I'd expect mathematically.